core/common/src/main/java/alluxio/wire/BackupStatus.java

```diff
@@ -0,0 +1,225 @@
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.wire;
+
+import alluxio.AlluxioURI;
+import alluxio.exception.AlluxioException;
+import alluxio.grpc.BackupPStatus;
+import alluxio.grpc.BackupState;
+
+import com.google.common.base.MoreObjects;
+import com.google.protobuf.ByteString;
+import org.apache.commons.lang.SerializationUtils;
+
+/**
+ * Defines the status of a backup.
+ */
+public class BackupStatus {
+  /** State of backup. */
+  private BackupState mState;
+  /** Host that is taking the backup. */
+  private String mBackupHost;
+  /** URI where the backup is located. */
+  private AlluxioURI mBackupUri;
+  /** Counter for how many entries are written in a backup. */
+  private long mEntryCount;
+  /** The error if backup failed. */
+  private AlluxioException mError;
+
+  /**
+   * Creates a new backup status.
+   *
+   * @param state state of the backup
+   */
+  public BackupStatus(BackupState state) {
+    mState = state;
+  }
+
+  /**
+   * Cloning constructor.
+   *
+   * @param status backup status to clone
+   */
+  public BackupStatus(BackupStatus status) {
+    mState = status.mState;
+    mBackupHost = status.mBackupHost;
+    mBackupUri = status.mBackupUri;
+    mEntryCount = status.mEntryCount;
+    mError = status.mError;
+  }
+
+  /**
+   * @param pStatus proto backup status
+   * @return backup status from proto representation
+   */
+  public static BackupStatus fromProto(BackupPStatus pStatus) {
+    BackupStatus status = new BackupStatus(pStatus.getBackupState());
+    if (pStatus.hasBackupHost()) {
+      status.setHostname(pStatus.getBackupHost());
+    }
+    if (pStatus.hasBackupUri()) {
+      status.setBackupUri(new AlluxioURI(pStatus.getBackupUri()));
+    }
+    if (pStatus.hasBackupError()) {
+      status.setError((AlluxioException) SerializationUtils
+          .deserialize(pStatus.getBackupError().toByteArray()));
+    }
+    status.setEntryCount(pStatus.getEntryCount());
+    return status;
+  }
+
+  /**
+   * @return current entry count in the backup
+   */
+  public long getEntryCount() {
+    return mEntryCount;
+  }
+
+  /**
+   * Sets entry count.
+   *
+   * @param entryCount the entry count
+   * @return the updated instance
+   */
+  public BackupStatus setEntryCount(long entryCount) {
+    mEntryCount = entryCount;
+    return this;
+  }
+
+  /**
+   * @return {@code true} if the backup is finished (completed | failed)
+   */
+  public boolean isFinished() {
+    return isCompleted() || isFailed();
+  }
+
+  /**
+   * @return {@code true} if the backup is completed successfully
+   */
+  public boolean isCompleted() {
+    return mState == BackupState.COMPLETED;
+  }
+
+  /**
+   * @return {@code true} if the backup is failed
+   */
+  public boolean isFailed() {
+    return mState == BackupState.FAULTED;
```

Comment:
Should the stated be `FAILED`?

Reply:
I could only make `Failed` as there is another enum in the same proto with value `FAILED`.

Reply:
Oh I see, most projects get around this by naming the enums like `BackupState.BACKUP_STATE_FAILED`

Reply:
I just didn't want to lengthen enum values. We can do that on the next 'Failed' value in the same file.